geo: set SRID to 4326 when using ST_GeomFromGeoJSON

otan opened the pull request:
This matches PostGIS 3.0 behavior.

Resolves #57104 

Release note (bug fix): ST_GeomFromGeoJSON now sets the SRID to 4326,
matching PostGIS 3.0 / RFC7946 behavior.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/57152)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57152#-:-MNO_7MN8vXK5FuRe0kE:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/geo.go, line 187 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/57152#-MNOZ97R1eB9ZPolGUF0:-MNOZ97R1eB9ZPolGUF1:browhyz) ([raw file](https://github.com/cockroachdb/cockroach/blob/84c2d8ef3d6dd4c5a6d0e8d825b7b72435bb1a1e/pkg/geo/geo.go#L187)):*
> ```Go
> // ParseGeometryFromGeoJSON parses the GeoJSON into a given Geometry.
> func ParseGeometryFromGeoJSON(json []byte) (Geometry, error) {
> 	// Note we set SRID to 4326 from here, to match PostGIS's behavior as per RFC7946.
> ```

can you add a link to the relevant section https://tools.ietf.org/html/rfc7946#section-4 


<!-- Sent from Reviewable.io -->

otan committed:
geo: set SRID to 4326 when using ST_GeomFromGeoJSON

This matches PostGIS 3.0 behavior.

Release note (bug fix): ST_GeomFromGeoJSON now sets the SRID to 4326,
matching PostGIS 3.0 / RFC7946 behavior.

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57152)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[pkg/geo/geo.go, line 187 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/57152#-MNOZ97R1eB9ZPolGUF0:-MNPdNTU1lkrQ9dCvLZS:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/84c2d8ef3d6dd4c5a6d0e8d825b7b72435bb1a1e/pkg/geo/geo.go#L187)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

can you add a link to the relevant section https://tools.ietf.org/html/rfc7946#section-4 
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola
